documents/README.md

```diff
@@ -0,0 +1,237 @@
+
+To register a new `DocumentWidget` we can use the API `addWidgetFactory` from the `DocumentRegistry`. The main arguments you need to add to the factory are the widget's name, the name of the model that this widget uses, a list of file types that the widget can open, and the list of file types that the widget is the default view.
+
+```ts
+// src/index.ts#L49-L67
+
+// Creating the widget factory to register it so the document manager knows about
+// our new DocumentWidget
+const widgetFactory = new ExampleWidgetFactory({
+  name: FACTORY,
+  modelName: 'example-model',
+  fileTypes: ['example'],
+  defaultFor: ['example']
+});
+
+// Add the widget to the tracker when it's created
+widgetFactory.widgetCreated.connect((sender, widget) => {
+  // Notify the instance tracker if restore data needs to update.
+  widget.context.pathChanged.connect(() => {
+    tracker.save(widget);
+  });
+  tracker.add(widget);
+});
+// Registering the widget factory
+app.docRegistry.addWidgetFactory(widgetFactory);
+```
+
+## Document Widget
+
+The `DocumentWidget` is the view that will open when opening the file. The `DocumentWidget` contains four main attributes:
+
+- `context`: The context is the bridge between the file on disk and its representation on the frontend. This context includes all the information about the file and some methods to handle the file as its content. Some other attributes you can find in the context are the `DocumentModel` and the `sessionContext`, which handles the communication with the backend.
+- `title`: Which handles the content of the tab.
+- `toolbar`: The editor's toolbar, where you can add different widgets to trigger actions on the document.
+- `contentHeader`: This is a panel between the toolbar and the main content area. You can see this header as a second toolbar or as a notification area.
+- `content`: The content is the main area of the `DocumentWidget` when you add the view for your document.
+
+## Document Model
+
+The `DocumentModel` represents the file in the frontend. Through the model, you can listen to changes in the state of the file like its metadata or some other properties like `dirty` that indicates that the content differs from disk, and you can modify and listen to changes on the content. The main methods on the `DocumentModel` are `toString` and `fromString`, every file but the notebook is loaded/saved to disk as a string using these methods.
+
+## Shared Model
+
+In JupyterLab v3.1, we introduced the package `@jupyterlab/shared-models` to swap `ModelDB` as a data storage to make the notebooks collaborative. We implemented these shared models using [Yjs](https://yjs.dev), a high-performance CRDT for building collaborative applications that automatically sync. You can find all the documentation of Yjs [here](https://docs.yjs.dev).
```

Comment:
Wondering whether we should also have a small disclaimer at the very top of the file to mention this is for JupyterLab 3.1 only?

Reply:
That's a great idea